taskvine/src/manager/vine_file_replica_table.c

```diff
@@ -51,6 +51,11 @@ int vine_file_replica_table_insert(struct vine_manager *m, struct vine_worker_in
 // remove a file from the remote file table.
 struct vine_file_replica *vine_file_replica_table_remove(struct vine_manager *m, struct vine_worker_info *w, const char *cachename)
 {
+	if (!w->current_files || !m->file_worker_table) {
```

Comment:
I think you want:  `if(!w) {` ?  If `(!w->current_files || !m->file_worker_table)` is ever true that sound like a bug we don't want to hide. On the other hand `if(!w) {` could be ok, if the worker went away but some structure kept a pointer to a replica.